File: 프로그래머스/unrated/181851. 전국 대회 선발 고사/전국 대회 선발 고사.java
import java.util.*;

class Solution {
    public int solution(int[] rank, boolean[] attendance) {
        Map<Integer, Integer> rankMap = new LinkedHashMap<>();
        for (int i = 0; i < rank.length; i++)
            if(attendance[i]) rankMap.put(rank[i], i);

        TreeMap<Integer, Integer> treeMap = new TreeMap<>(rankMap);
        ArrayList<Integer> arrayList = new ArrayList<>(treeMap.values());

        return arrayList.get(0) * 10000
             + arrayList.get(1) * 100
             + arrayList.get(2);
    }
}

File: 프로그래머스/unrated/181851. 전국 대회 선발 고사/전국 대회 선발 고사.py
def solution(rank, attendance):
    rankArr = []
    rankDict = {}
    for i in range(len(rank)):
        rankDict[rank[i]] = i
        if attendance[i]:
            rankArr.append(rank[i])
            
    rankArr.sort()
    
    return rankDict[rankArr[0]] * 10000 + rankDict[rankArr[1]] * 100 + rankDict[rankArr[2]]